docroot/modules/humsci/hs_dashboard/src/AnnouncementsManager.php

```diff
@@ -0,0 +1,299 @@
+   *   The date formatter interface.
+   */
+  public function __construct(
+    ClientInterface $http_client,
+    LoggerChannelFactoryInterface $logger_factory,
+    FileSystemInterface $file_system,
+    DateFormatterInterface $date_formatter,
+  ) {
+    $this->httpClient = $http_client;
+    $this->logger = $logger_factory->get('hs_dashboard');
+    $this->fileSystem = $file_system;
+    $this->dateFormatter = $date_formatter;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('http_client'),
+      $container->get('logger.factory'),
+      $container->get('file_system'),
+      $container->get('date.formatter'),
+    );
+  }
+
+  /**
+   * Retrieves the CSV from Google Sheets.
+   *
+   * @param string $url
+   *   The URL of the CSV file to retrieve.
+   *
+   * @return array
+   *   Returns a csv array - if one is not found, an empty array is returned.
+   */
+  private function getCsvAnnouncements($url): array {
+    try {
+      $response = $this->httpClient->request('GET', $url, [
+        'headers' => [
+          'Accept' => 'text/csv',
+        ],
+        'timeout' => 10,
+      ]);
+
+      if ($response->getStatusCode() !== 200) {
+        $this->logger->error('Invalid response status from {url} { message }: ', [
+          'url' => $url,
+          'message' => $response->getStatusCode(),
+        ]);
+        throw new \Exception('Invalid response status: ' . $response->getStatusCode());
+      }
+
+      $csv_content = $response->getBody()->getContents();
+      return $this->parseCsv($csv_content);
+
+    }
+    catch (RequestException $e) {
+      $this->logger->error('Error retrieving CSV from {url}: {message}', [
+        'url' => $url,
+        'message' => $e->getMessage(),
+      ]);
+      return [];
+    }
+  }
+
+  /**
+   * Parses a CSV file.
+   *
+   * @param string $csv_content
+   *   The CSV content.
+   *
+   * @return array
+   *   An array of CSV data.
+   */
+  private function parseCsv(string $csv_content): array {
+    $rows = [];
+
+    $temp_file_path = 'temporary://csv_import.csv';
+
+    $file_uri = $this->fileSystem->saveData($csv_content, $temp_file_path, FileExists::Replace);
+
+    if ($file_uri) {
+      $handle = fopen($this->fileSystem->realpath($file_uri), 'r');
+
+      if ($handle !== FALSE) {
+        // Skip first header row.
+        $first_row = TRUE;
+        while (($data = fgetcsv($handle)) !== FALSE) {
+          if ($first_row) {
+            $first_row = FALSE;
+            continue;
+          }
+
+          // Removes empty rows.
+          if (empty($data[2]) && empty($data[3])) {
+            continue;
+          }
+
+          if (isset($data[1])) {
+            $data[1] = $this->convertDateToTimestamp(trim($data[1]));
+          }
+
+          if (isset($data[2])) {
+            $data[2] = $this->convertMarkdownLinks('<strong>' . trim($data[2]) . '</strong>');
```

Comment:
Sorry, I missed this in the previous review, but I think it's better to add the `<strong>` after doing the markdown conversion. I think (not 100% sure) you can add a `#prefix` and a `#suffix` to the render array in line 289 or just do the concatenation there, as all the titles have it and it doesn't affect the markdown parsing.

Reply:
Fixed.